### PR TITLE
11 feature 11 result pulling and extraction

### PR DESCRIPTION
**Pull request by Jerryfish0807**
Fully establish the feature of exporting but lacks peer-reviewed. (Java Doc has been entirely written, please use it as a reference to test)

### CODE REVIEW

**Review by j-y26 — CHANGES_REQUESTED** [review]
This is in general a very good use case that satisfies the specification. However, please do refactor many parts I commented into helper methods. In addition, this use case should use a Builder Design pattern.  Thus, for the methods mentioned previously about refactoring, refactor them into another class called StudyResulltBuilder and define the aforementioned methods. Inject this builder class into the interactor class so that your interactor uses these methods in the same process where you are creating the csv files now.

**Review by Tedward765 — APPROVED** [review]
:)

File: src/main/java/use_cases/result_extraction/ResultExtractionInteractor.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+     * <p>
+     *
+     * @param studyID The ID of the study that the researcher want to export
+     * @param filepath The file path to save the result folder
+     */
+
+    @Override
+    public void resultPullingAndExtraction(int studyID, String filepath) {
+        Study study = FetchId.getStudy(studyID);
+        ResultExtractionBuilder fBuilder = new ResultExtractionBuilder();
```
Comment (j-y26) [inline src/main/java/use_cases/result_extraction/ResultExtractionInteractor.java:53]: The hard dependency here can be made to dependency injection

**Review by j-y26 — APPROVED** [review]
This version of the result extraction use case has made significant improvements compared to the previous version by incorporating the use of a Builder design pattern. The interaction now does not contain any code smells with a step-by-step process to call the builder to build the "saved results". 

One small thing that can be improved is to use dependency injection to inject the builder into the interactor, without directly initializing the builder with the new method. However, this is minor and the use case is good enough to be merged to main.

**Review by j-y26 — DISMISSED** [review]

**Review by Tedward765 — APPROVED** [review]
:)

**Review by xuyangan — DISMISSED** [review]
Nice job on using the builder pattern instead of a switch block.

**Review by xuyangan — APPROVED** [review]
I deleted the conflicts so that the branch can be merged to main.

**Review by j-y26 — APPROVED** [review]
Conflict resolved. Conflict resolved

**Review by xustell — APPROVED** [review]
okay!

**Pull request merged**